lib/update/cluster/builder.go

```diff
@@ -317,24 +317,40 @@ func (r phaseBuilder) masters(leadMaster storage.UpdateServer, otherMasters []st
 			}})
 
 		// election - stepdown first node we will upgrade
-		node.AddSequential(setLeaderElection(enable(), disable(leadMaster), leadMaster, "stepdown", "Step down %q as Kubernetes leader"))
+		node.AddSequential(setLeaderElection(
+			electionChanges{
+				id:          "stepdown",
+				description: fmt.Sprintf("Step down %q as Kubernetes leader", leadMaster.Hostname),
+				disable:     serversToStorage(leadMaster),
+			},
+			leadMaster,
+		))
+
+		// election - force failover to first upgraded master
+		electionChanges := electionChanges{
+			description: fmt.Sprintf("Make node %q Kubernetes leader", leadMaster.Hostname),
+			enable:      serversToStorage(leadMaster),
+			disable:     serversToStorage(otherMasters...),
+		}
+
+		node.AddSequential(r.commonNode(leadMaster, leadMaster, supportsTaints,
+			waitsForEndpoints(len(otherMasters) == 0), electionChanges)...)
```

Comment:
Nit: Since you moved this under `if len(otherMasters) != 0`, here and below you can just pass bool flags which will improve readability a little bit, i.e. `waitForEndpoints(false)` here and `waitForEndpoints(true)` below.